app/src/ui/dialog/dialog.tsx

```diff
@@ -187,6 +196,31 @@ export class Dialog extends React.Component<IDialogProps, IDialogState> {
 
     this.setState({ isAppearing: true })
     this.scheduleDismissGraceTimeout()
+
+    window.addEventListener('focus', this.onWindowFocus)
+  }
+
+  private onWindowFocus = () => {
+    // On Windows, a click which focuses the window will also get passed down
+    // into the DOM. But we don't want to dismiss the dialog based on that
+    // click. See https://github.com/desktop/desktop/issues/2486.
+    if (__WIN32__) {
+      this.clearClickDismissalTimer()
+
+      this.disableClickDismissal = true
+
+      setTimeout(() => {
```

Comment:
We haven't really been consistent here with how we use `setTimeout` - sometime's it explicitly off `window`, sometimes the global scope is inferred - this feels fine but then you get TYPES coming along and tearing me apart:

<img width="683" alt="screen shot 2017-08-21 at 1 31 36 pm" src="https://user-images.githubusercontent.com/359239/29502687-1ec6890c-8675-11e7-83cd-57ee4d4d9a46.png">

Basically one set of API returns an `NodeJS.Timer` or a numeric handle. I'm going to push a commit to make this explicitly `window.setTimeout` because we weren't assigning this result value to `this.disableClickDismissalTimeoutId` and so `clearTimeout` wasn't ever used.